lib/compile/util.js

```diff
@@ -214,20 +214,33 @@ function getPath(currentPath, prop, jsonPointers) {
 }
 
 
+var JSON_POINTER = /^#|\/(?:[^~]|~0|~1)*?$/;
```

Comment:
`/defenitions/foo` is a valid relative URI (definitions is a folder and foo is a file), schema compilation fails because it cannot find schema with such uri.

But `#/foo` is not a valid JSON pointer, it passes json-pointer format validation because I allowed (maybe mistakenly? Not sure, it's not precisely specified) URI encoded JSON-pointers in this format.

I don't think it's such a big deal to worry about, as long as you are aware that $data only takes normal JSON pointers it's fine. Allowing uri encoded JSON pointers would be a bit more effort that just stripping #, you'd need to decode as well.